test/python/ph/test_data_collection.py

```
@@ -0,0 +1,132 @@
+        setups = build_setup_grid([4], [3])
+        assert setups == []
+
+    @staticmethod
+    def test_valid_single_setup():
+        """Test that a single valid (num_modes, target_dim) pair produces one Setup."""
+        setups = build_setup_grid([4], [2])
+        assert len(setups) == 1
+        assert setups[0] == Setup(num_modes=4, target_dim=2)
+
+    @staticmethod
+    def test_valid_multiple_setups():
```

Ruff (0.15.20)

[warning] Missing return type annotation for staticmethod `test_valid_multiple_setups`

Add return type annotation: `None`

(ANN205)